src/components/PlantsPrimaryPage/PlantsPrimaryPageView.tsx

```diff
@@ -54,17 +54,30 @@ export default function PlantsPrimaryPageView({
   const contentRef = useRef(null);
   const { isAcceleratorRoute } = useAcceleratorConsole();
 
+  const isRolledUpView = useMemo(() => {
+    return projectId !== undefined && selectedPlantingSiteId === -2;
+  }, [projectId, selectedPlantingSiteId]);
+
   const onChangePlantingSiteId = (siteId: any) => {
     const selectedPlantingSite = plantingSites?.find((ps) => ps.id === siteId);
     if (selectedPlantingSite) {
       onSelect(selectedPlantingSite);
+    } else if (siteId === -2) {
+      onSelect(undefined);
     }
   };
 
-  const options = useMemo(
-    () => plantingSites?.map((site) => ({ label: site.name, value: site.id })) ?? [],
-    [plantingSites]
-  );
+  const options = useMemo(() => {
+    const optionsToReturn = plantingSites?.map((site) => ({ label: site.name, value: site.id })) ?? [];
+    if (projectId) {
+      optionsToReturn.unshift({ label: strings.ALL_PLANTING_SITES, value: -2 });
+    }
```

Comment:
Should this only `unshift` if `optionsToReturn.length !== 1`?

Reply:
Yes, if we only have one site we shouldn't add the rolled up view. I was not including that condition yet, but I can add it